holoviews/core/spaces.py

```diff
@@ -513,6 +515,25 @@ def __call__(self, *args, **kwargs):
         return ret
 
 
+
+class Generator(Callable):
+    """
+    Generators are considered a special case of Callable that accept no
+    arguments and never memoize.
+    """
+
+    callable = param.Parameter(default=None, constant=True, doc="""
+         The generator function being wrapped.""")
```

Comment:
I don't believe param has a ``Generator`` parameter which is why I am using ``param.Parameter``. Maybe I could use a class selector over generator types?